File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        summ = sum(arr[:k])
        avg = summ/k 
        count = 0
        if avg >= threshold : count+=1
        for i in range(k, len(arr)) : 
            summ+=arr[i]
            summ-=arr[i-k]
            avg = summ/k 
            if avg >= threshold : count+=1
        return count

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @param {number} threshold
 * @return {number}
 */
var numOfSubarrays = function(arr, k, threshold) {
    let res = 0, sum = 0
    for(let i = 0; i<k; i++){
        sum+=arr[i]
    }
    if ((sum / k) >= threshold) res+=1
    for (let i = k; i<arr.length; i++){
        sum+=arr[i]
        sum-=arr[i-k]
        if ((sum / k) >= threshold) res+=1
    }
    return res
};